From 2188f6835608967a98ff99d5c841f98c62a907ae Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Wed, 6 Jan 2016 18:40:31 +0800
Subject: [PATCH 3635/5242] MLK-12093-2 ARM: imx: correct coding error

commit  487ae280512a1893b946d2458c3104822fe17453 from
https://source.codeaurora.org/external/imx/linux-imx.git

Correct coding error and use macro define instead
of register value.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/lpddr3_freq_imx.S |    6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-imx/lpddr3_freq_imx.S b/arch/arm/mach-imx/lpddr3_freq_imx.S
index 624fc4e..a33b893 100644
--- a/arch/arm/mach-imx/lpddr3_freq_imx.S
+++ b/arch/arm/mach-imx/lpddr3_freq_imx.S
@@ -23,7 +23,7 @@
 #define DDRC_PSTAT		0x3fc
 #define DDRC_PCTRL_0		0x490
 #define DDRC_DFIMISC		0x1b0
-#defien DDRC_DBGCAM		0x308
+#define DDRC_DBGCAM		0x308
 #define DDRC_SWCTL		0x320
 #define DDRC_SWSTAT		0x324
 #define DDRPHY_LP_CON0		0x18
@@ -76,10 +76,8 @@
 	str	r7, [r4, #DDRC_DBG1]
 
 	ldr	r6, =0x30000000
-	ldr	r8, =0x300
-	orr	r8, r8, #0x8
 3:
-	ldr	r7, [r4, r8]
+	ldr	r7, [r4, #DDRC_DBGCAM]
 	and	r7, r7, r6
 	cmp	r7, r6
 	bne	3b
-- 
1.7.9.5

